src/client/client.ts

```diff
@@ -14,12 +14,60 @@ interface ErrorWithAdditionalInfo extends Error {
   statusCode: number;
 }
 
-export interface HTTPClientResponse {
-  body: Uint8Array | any; // when content-type=JSON, body is a JSON object, otherwise it's a Uint8Array
+export class HTTPClientResponse {
+  /**
+   * The raw response bytes
+   */
+  body: Uint8Array;
+  /**
+   * If the expected response type is JSON, this is the response bytes converted to a string.
+   */
   text?: string;
+  format: 'application/msgpack' | 'application/json';
   headers: Record<string, string>;
   status: number;
   ok: boolean;
+
+  constructor(options: {
+    body: Uint8Array;
+    text?: string;
+    format: 'application/msgpack' | 'application/json';
+    headers: Record<string, string>;
+    status: number;
+    ok: boolean;
+  }) {
+    this.body = options.body;
+    this.text = options.text;
+    this.format = options.format;
+    this.headers = options.headers;
+    this.status = options.status;
+    this.ok = options.ok;
+  }
+
+  /**
+   * Returns the response body as a string, ready to be parsed as JSON.
+   */
+  getJSONText(): string {
```

Comment:
NIT: Why have the word JSON in the funcion name at all unless checking the content type?

Reply:
`this.text` is only populated if the response is JSON. This happens on line 232